cardano-api/src/Cardano/Api/Fees.hs

```diff
@@ -787,16 +787,25 @@ makeTransactionBodyAutoBalance eraInMode systemstart history pparams
                        | Map.null failures -> return exUnitsMap'
                        | otherwise         -> Left (TxBodyScriptExecutionError
                                                      (Map.toList failures))
-
-    let txbodycontent1 = substituteExecutionUnits exUnitsMap' txbodycontent
-
     explicitTxFees <- first (const TxBodyErrorByronEraNotSupported) $
                         txFeesExplicitInEra era'
 
+    let txbodycontent1 = substituteExecutionUnits exUnitsMap' txbodycontent {
+                         txFee  = TxFeeExplicit explicitTxFees 0,
+                         txOuts = TxOut changeaddr
+                                        (lovelaceToTxOutValue 0)
+                                        TxOutDatumHashNone
+                                : txOuts txbodycontent
+                         }
+
     -- Insert change address and set tx fee to 0
     txbody1 <- first TxBodyError $ -- TODO: impossible to fail now
                makeTransactionBody txbodycontent1 {
-                 txFee = TxFeeExplicit explicitTxFees 0
+                 txFee = TxFeeExplicit explicitTxFees 0,
+                 txOuts = TxOut changeaddr
+                                (lovelaceToTxOutValue 0)
+                                TxOutDatumHashNone
+                        : txOuts txbodycontent
```

Comment:
This bit isn't necessary since it's overriding `txbodycontent1` which already has those values, not the original `txbodycontent`.

Reply:
Yep its not, I wanted to put something out asap for QA. Fixing now.